netboot-services/ipxeMenuGenerator/main_test.go

```diff
@@ -82,3 +176,207 @@ func (m *mockFileInfo) IsDir() bool {
+goto startboot
+
+
+######################
+# Development-Images #
+######################
+
+
+:thinclient-24-08-08-noissue-needmoreram-5f1a800
+set squash_url ${http-protocol}://${basicAuth}${url}/dev/24-08-08-noissue-needmoreram-5f1a800/dg-thinclient.squashfs${sas_token}
+set kernel_url ${http-protocol}://${basicAuth}${url}/dev/24-08-08-noissue-needmoreram-5f1a800/
+goto startboot-dev
+`
+
+	setupTestFiles(mockFileNameTemplate, []byte(mockFileContentTemplate))
+	defer cleanupTestFiles([]string{mockFileNameTemplate, mockFileNameGenerated})
+
+	prodImages, err := getImages(ProdFolder)
+	if err != nil {
+		log.Error(err)
+	}
+
+	devImages, err := getImages(DevFolder)
+	if err != nil {
+		log.Error(err)
+	}
+
+	ExampleDataToRender.JinjaTemplateFile = mockFileNameTemplate
+
+	err = renderAdvancedMenu(ExampleDataToRender, prodImages, devImages)
+	assert.NilError(t, err)
+
+	fileContentGenerated, err := os.ReadFile(mockFileNameGenerated)
+	assert.Equal(t, string(fileContentGenerated), mockFileContentGenerated)
+	assert.NilError(t, err)
+}
+
+// For sake of code coverage, we are testing renderNetinfoMenu() function. It basically just rewrites the file with the same content as we do not pass custom data to it.
+func TestRenderNetinfoMenu(t *testing.T) {
+	mockFileNameTemplate := "netinfo_test.ipxe.j2"
+	mockFileNameGenerated := "netinfo_test.ipxe"
+	mockFileContentTemplate := `#!ipxe
+menu Network info
+item --gap MAC:
+item mac ${sp} ${netX/mac}
+item --gap IP/mask:
+item ip ${sp} ${netX/ip}/${netX/netmask}
+item --gap Gateway:
+item gw ${sp} ${netX/gateway}
+item --gap Domain:
+item domain ${sp} ${netX/domain}
+item --gap DNS:
+item dns ${sp} ${netX/dns}
+item --gap DHCP server:
+item dhcpserver ${sp} ${netX/dhcp-server}
+item --gap Next-server:
+item nextserver ${sp} ${next-server}
+item --gap Filename:
+item filename ${sp} ${netX/filename}
+choose empty ||
+exit`
+
+	mockFileContentGenerated := `#!ipxe
+menu Network info
+item --gap MAC:
+item mac ${sp} ${netX/mac}
+item --gap IP/mask:
+item ip ${sp} ${netX/ip}/${netX/netmask}
+item --gap Gateway:
+item gw ${sp} ${netX/gateway}
+item --gap Domain:
+item domain ${sp} ${netX/domain}
+item --gap DNS:
+item dns ${sp} ${netX/dns}
+item --gap DHCP server:
+item dhcpserver ${sp} ${netX/dhcp-server}
+item --gap Next-server:
+item nextserver ${sp} ${next-server}
+item --gap Filename:
+item filename ${sp} ${netX/filename}
+choose empty ||
+exit`
+
+	setupTestFiles(mockFileNameTemplate, []byte(mockFileContentTemplate))
+	defer cleanupTestFiles([]string{mockFileNameTemplate, mockFileNameGenerated})
+
+	ExampleDataToRender.JinjaTemplateFile = mockFileNameTemplate
+
+	err := renderNetinfoMenu(ExampleDataToRender)
+	assert.NilError(t, err)
+
+	fileContentGenerated, err := os.ReadFile(mockFileNameGenerated)
+	assert.Equal(t, string(fileContentGenerated), mockFileContentGenerated)
+	assert.NilError(t, err)
+
+}
+
+// Test getMostRecentSquashfsImage() --> Tested with TestRenderMenu
```

Comment:
Die Funktion wird zwar in einem Unittest aufgerufen, aber ob die auch etwas sinnvolles tut, wird nicht überprüft.